File: java/src/test/java/fr/eulbobo/dojo/byron/domain/ByronEventFactoryTest.java
package fr.eulbobo.dojo.byron.domain;

import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

import java.util.stream.Stream;

import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.Mockito.mock;

class ByronEventFactoryTest {

    static Stream<Arguments> byronEvents() {
        return Stream.of(Arguments.of(1, ByronRecreation.class),
                Arguments.of(2, ByronRecreation.class),
                Arguments.of(3, ByronDrama.class),
                Arguments.of(4, ByronDrama.class),
                Arguments.of(5, BriefRedoubt.class),
                Arguments.of(6, BriefRedoubt.class));
    }

    @ParameterizedTest
    @MethodSource("byronEvents")
    void shouldGetCorrectByronEventOnDiceRoll(int roll, Class expectedInterface) {
        ByronEventFactory factory = new ByronEventFactory(mock(EventLog.class));
        assertThat(factory.from(DiceRoll.of(roll))).isExactlyInstanceOf(expectedInterface);
    }
}

File: c#/fr.eulbobo.dojo.byron/fr.eulbobo.dojo.byron.tests/ByronEventFactoryTest.cs
using fr.eulbobo.dojo.byron.domain;
using Moq;

namespace fr.eulbobo.dojo.byron.tests
{
    public class ByronEventFactoryTest
    {

        [TestCase(1, typeof(ByronRecreation))]
        [TestCase(2, typeof(ByronRecreation))]
        [TestCase(3, typeof(ByronDrama))]
        [TestCase(4, typeof(ByronDrama))]
        [TestCase(5, typeof(BriefRedoubt))]
        [TestCase(6, typeof(BriefRedoubt))]
        public void ShouldGetCorrectByronEventOnDiceRoll(int roll, Type expectedInterface)
        {
            var eventLog = new Mock<EventLog>();
            ByronEventFactory factory = new ByronEventFactory(eventLog.Object);
            Assert.IsInstanceOf(expectedInterface, factory.From(DiceRoll.Of(roll)));
        }
    }
}
